src/common/snippets/src/runtime_configurator.cpp

```diff
@@ -272,9 +259,182 @@ void RuntimeConfigurator::update_data_offsets() const {
+    }
+}
+
+bool RuntimeConfigurator::ParallelWAOptimizer::need_optimize(const VectorDims& master_shape) {
+    return !loops_to_split.empty() && SplitDimensionM::split(master_shape, concurrency, batch_m, new_m);
+}
+
+void RuntimeConfigurator::ParallelWAOptimizer::update(RuntimeConfigurator::LoopInfoRuntimeParamsMap& map,
+                                                      std::vector<VectorDims>& shapes,
+                                                      std::vector<std::vector<size_t>>& layouts,
+                                                      const std::shared_ptr<RuntimeConfig>& config,
+                                                      size_t in_num) {
+    update_split_loops_info(map);
+    update_shapes(shapes);
+    update_layouts(layouts);
+    update_config(config);
+}
+
+void RuntimeConfigurator::ParallelWAOptimizer::update_split_loops_info(LoopInfoRuntimeParamsMap& initialized_info) {
+    for (const auto& loop : loops_to_split) {
+        if (initialized_info.count(loop) == 0) {
+            loop->set_work_amount(new_m);
+            lowered::pass::InitLoops::update_data_pointer_shifts(loop);
+            initialized_info[loop] = compute_runtime_params(loop);
+        }
+    }
+}
+
+void RuntimeConfigurator::ParallelWAOptimizer::update_shapes(std::vector<VectorDims>& shapes) {
+    for (size_t i = 0; i < m_dim_idces.size(); ++i) {
+        shapes[i] = not_m_related_params.count(i)
+                        ? SplitDimensionM::unsqueeze_m_dim(shapes[i], m_dim_idces[i])
+                        : SplitDimensionM::reshape_m_dim(shapes[i], m_dim_idces[i], batch_m, new_m);
+    }
+}
+
+void RuntimeConfigurator::ParallelWAOptimizer::update_layouts(std::vector<std::vector<size_t>>& layouts) {
+    layouts = split_layouts;
+}
+
+void RuntimeConfigurator::ParallelWAOptimizer::update_config(const std::shared_ptr<RuntimeConfig>& config) {
+    *++config->master_shape.rbegin() = new_m;
+    config->master_shape.insert(config->master_shape.cbegin() + config->master_shape.size() - 2, batch_m);
+    config->tensor_rank = config->master_shape.size();
+}
+
+std::unordered_set<ExpressionPtr> RuntimeConfigurator::ParallelWAOptimizer::find_applicable_brgemms(
+    const lowered::LinearIRPtr& linear_ir) {
+    auto is_brgemm = [](const ExpressionPtr& expr) {
+        return ov::is_type<op::Brgemm>(expr->get_node());
+    };
+    auto brgemm_it = std::find_if(linear_ir->begin(), linear_ir->end(), is_brgemm);
+    std::unordered_set<ExpressionPtr> brgemms;
+    while (brgemm_it != linear_ir->end()) {
+        brgemms.insert(*brgemm_it);
+        brgemm_it = std::find_if(std::next(brgemm_it), linear_ir->end(), is_brgemm);
+    }
+    const auto& loop_manager = linear_ir->get_loop_manager();
+    auto applicable_brgemm = [&loop_manager](const ExpressionPtr& expr) {
+        const auto& loop_idces = expr->get_loop_ids();
+        if (loop_idces.empty())
```

Comment:
Could you please add the comment which brgemms are applicable?
Something like: `Brgemms` are in Loop by M with defined work amount. We have to have Loop by M due to (? since executor::update_config() takes dimension M from shape in case without Loop by M?)

Reply:
The comment just before the lambda is added